src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -81,11 +86,14 @@ public AssemblyScannerResults GetScannableAssemblies()
 
             var assemblies = new List<Assembly>();
 
-            foreach (var assemblyFile in ScanDirectoryForAssemblyFiles(baseDirectoryToScan, ScanNestedDirectories))
+            foreach (var directoryToScan in baseDirectoriesToScan)
```

Comment:
this could indicate that the order of directories scanned is not relevant. However, looking at https://docs.microsoft.com/en-us/dotnet/api/system.reflection.assembly.loadfrom?view=netcore-3.1 I'm not sure this is absolutely safe e.g. in scenarios where the additional scanning path contains an assembly with the same identity as in the default path (which should be part of the probing path) and the additional path would be scanned first.

> If an assembly with the same identity is already loaded, LoadFrom returns the loaded assembly even if a different path was specified.

This seems to be very theoretical though and at least for now, it's probably safe to assume that the scanning happens in the expected order.